internals/overlord/planstate/manager.go

```diff
@@ -48,6 +48,12 @@ func NewManager(layersDir string) (*PlanManager, error) {
 	return manager, nil
 }
 
+// RegisterExtension adds a plan extension which extends the plan schema. All
+// extensions must be registered before Load() is called.
+func (m *PlanManager) RegisterExtension(field string, ext plan.LayerSectionExtension) error {
```

Comment:
This makes it look like registering an extension is tied to the plan manager, but it's really global. I think it'd be better for callers just to use `plan.RegisterExtension` directly so that that's clear.

Reply:
Makes sense, thanks. Applied. Resolving.